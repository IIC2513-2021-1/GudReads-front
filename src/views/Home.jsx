import React from 'react';
import { Link } from 'react-router-dom';
import useAuth from '../hooks/useAuth';

export default function Home() {
  const { currentUser, handleUserLogout } = useAuth();

  return (
    <div>
      <h2>
        Welcome to Gud Reads
        {' '}
        {currentUser?.firstName}
      </h2>
      <Link to="/authors">Authors</Link>
      <div>
        {currentUser ? (
          <button type="button" onClick={handleUserLogout}>Logout</button>
        ) : (
          <>
            <h3>Have an account? Log In</h3>
            <Link to="/login">Login</Link>
            <h3>Do not have an account yet? Sign Up</h3>
            <Link to="/signup">Sign up!</Link>
          </>
        )}
      </div>
    </div>
  );
}
