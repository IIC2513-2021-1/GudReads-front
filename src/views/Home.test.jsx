import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import AuthContextProvider from '../contexts/AuthContext';
import Home from './Home';

function TestWrapper({ children, path }) {
  return (
    <MemoryRouter initialEntries={[path]}>
      <AuthContextProvider>
        {children}
      </AuthContextProvider>
    </MemoryRouter>
  );
}

const user = {
  firstName: 'Jalen',
  lastName: '87',
  email: 'Jalen83@gmail.com',
  access_token: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOjIyLCJpYXQiOjE2MjU0OTk0ODEsImV4cCI6MTYyNTUwMTI4MX0.jXD-n92FpFo6zABoRykc8FdKcRoisKhUl0jcTPrc31E',
  token_type: 'Bearer',
};

const sessionExpiration = new Date(
  new Date().getTime() + 1000 * 60 * 60 * 24, // One day in the future
);

const localStorageMapping = {
  user,
  sessionExpiration,
};

describe('Home', () => {
  describe('when user is not logged in', () => {
    it('renders the home page with log in link', () => {
      const tree = renderer.create(<TestWrapper path="/"><Home /></TestWrapper>).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('when user is logged in', () => {
    beforeEach(() => {
      global.Storage.prototype.getItem = jest.fn(
        (key) => JSON.stringify(localStorageMapping[key]),
      );
    });

    afterEach(() => {
      global.Storage.prototype.getItem.mockReset();
    });

    it('renders the home page with logout button', () => {
      const tree = renderer.create(<TestWrapper path="/"><Home /></TestWrapper>).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
