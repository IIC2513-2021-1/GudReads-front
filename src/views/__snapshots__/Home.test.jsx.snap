// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home when user is logged in renders the home page with logout button 1`] = `
<div>
  <h2>
    Welcome to Gud Reads
     
    Jalen
  </h2>
  <a
    href="/authors"
    onClick={[Function]}
  >
    Authors
  </a>
  <div>
    <button
      onClick={[Function]}
      type="button"
    >
      Logout
    </button>
  </div>
</div>
`;

exports[`Home when user is not logged in renders the home page with log in link 1`] = `
<div>
  <h2>
    Welcome to Gud Reads
     
  </h2>
  <a
    href="/authors"
    onClick={[Function]}
  >
    Authors
  </a>
  <div>
    <h3>
      Have an account? Log In
    </h3>
    <a
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
    <h3>
      Do not have an account yet? Sign Up
    </h3>
    <a
      href="/signup"
      onClick={[Function]}
    >
      Sign up!
    </a>
  </div>
</div>
`;
